Update Eclipse dependency to 4.25

bjhargrave opened the pull request:
Move from Eclipse 4.16 to Eclipse 4.25 as the base level of Eclipse.

bjhargrave commented:
@timothyjward I hope to have properly changed the m2e code for m2e 2.x, but please check to see if more is needed.

timothyjward commented:
> @timothyjward I hope to have properly changed the m2e code for m2e 2.x, but please check to see if more is needed.

I have added some additional commits in https://github.com/bjhargrave/bnd/pull/46 which can either be merged into this PR, or added as a separate PR in this repository

bjhargrave commented:
> I have added some additional commits in [bjhargrave#46](https://github.com/bjhargrave/bnd/pull/46) which can either be merged into this PR

Thanks! I squashed your fixes into the main commit and force pushed here.
